[Ng Wei En] iP

wei2912 opened the pull request:
# Duke

> Quote text here

## Lorem ipsum blah blah

Here's a *bullet* list:
* abc1
* def2
* ghi3

Here's a **numbered** list:
1. foo
2. bar
3. foobar

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("hello world!");
    }
}
```

Todo:
- [ ] Feature 1
- [ ] Feature 2
- [ ] Feature 3

Reference to `duke`: https://github.com/nus-cs2103-AY2122S1/ip :P

wei2912 committed:
Implement skeleton of Duke

wei2912 committed:
Duke: Add some extra whitespace

wei2912 committed:
Implement store text and list functions

wei2912 committed:
Isolate list preparation code into function

wei2912 committed:
Implement done marking

wei2912 committed:
Implement Todo/Deadline/Event tasks

wei2912 committed:
Fix bug whith repeated task toggles

wei2912 committed:
.gitignore: Correct typo

wei2912 committed:
Update tests with current expected output

wei2912 committed:
Duke: Add rudimentary error handling

wei2912 committed:
Duke: Implement delete command

wei2912 committed:
Duke: Rename functions

wei2912 committed:
Duke: Refactor command parsing

wei2912 committed:
Duke: Correct typo

wei2912 committed:
Duke: Implement read/write file

wei2912 committed:
Implement support for dates

wei2912 committed:
Merge branch 'branch-Level-7'

wei2912 committed:
Merge branch 'branch-Level-8'

wei2912 committed:
Duke: Separate into classes

wei2912 committed:
Package files as duke

wei2912 committed:
Set up Gradle

wei2912 committed:
Parser: Add tests

wei2912 committed:
Gradle: Add Shadow plugin

wei2912 committed:
Document code

wei2912 committed:
Amend code to fit coding standard

wei2912 committed:
Implement find command

wei2912 committed:
Merge branch 'branch-A-JavaDoc'

wei2912 committed:
Merge branch 'branch-A-CodingStandard'

wei2912 committed:
Merge branch 'branch-Level-9'

wei2912 committed:
Set up CheckStyle

wei2912 committed:
Fix coding standard violations

wei2912 committed:
Merge branch 'branch-A-CheckStyle'

wei2912 committed:
Change modes of files

wei2912 committed:
.gitignore: Ignore data folder

sunjc826 commented on src/main/java/duke/TaskType.java:
```diff
@@ -0,0 +1,56 @@
+        this.taskIcon = taskIcon;
+    }
+
+    /**
+     * Getter for task icon.
+     * @return Task icon corresponding to type of task.
+     */
```
I'm not so sure if documentation for getters is necessary. To compare it with your `TaskList` class, there are trivial methods there that are undocumented. Hence, to be consistent, I suggest, either adding documentation to those methods as well, or removing documentation for methods whose purpose is clear from the name.

feliciaivane replied:
I suggest adding documentation to the methods in the other classes as well 😄

wei2912 replied:
I'll work on that in future iterations after I reorganise my code for the GUI, thanks for the suggestions!

sunjc826 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,153 @@
+     * @return Input task rendered as a string.
+     */
+    public static String renderTask(Task task) {
+        var taskType = TaskType.identifyTask(task);
```
Interesting use of `var`. :) (and elsewhere in your code)
To give another perspective of this however, without an ide, say I'm viewing the codebase on github, I think it would be harder to tell what is the type of the variable. Hence, I'm not so sure if this is the "Java" way. I suppose this is still way better than reading raw code from many dynamically typed languages out there.
Regardless, viewing code without an ide probably isn't a major usecase these days so this shouldn't be an issue.

wei2912 replied:
I'm not sure if `var` is permissible nowadays and the points you mentioned are certainly valid. I don't think it's a coding standard violation and I still can't find any standards that recommend for/against it, so I figured I'd just use it whenever I can to avoid having to put the types in 😂 Let me know if you find out more!

sunjc826 commented on src/main/java/duke/TaskType.java:
```diff
@@ -0,0 +1,56 @@
+     * @param task Task.
+     * @return Type of input task.
+     */
+    public static TaskType identifyTask(Task task) {
```
Interesting decoupling of domain logic and cli presentation.

sunjc826 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+    private LocalDate deadline;
+
+    /**
+     * Constructor for Deadline object.
```
```suggestion
     * Constructor for Deadline object.
 
```
And for other methods as well. There should be a newline between description and params.

feliciaivane replied:
Yes, I agree with this.

wei2912 replied:
Will note that when reworking my documentation, thanks for pointing it out!

sunjc826 reviewed (commented):
Excellent work! Other than some newlines here and there, there aren't many nitpicks I can find.

Also, this is a first I've seen the use of var in a Java project. Very interesting.

feliciaivane commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,112 @@
+            );
+        }
+        return false;
+    }
```
I like how there is no deep nesting present in this method, but it is rather long. Would it be possible to split this method up into several shorter ones? :D

wei2912 replied:
Good point! I plan to add some additional functions to reduce the amount of repetition, and perhaps create `Command` classes for this purpose.

feliciaivane commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,153 @@
+                        i + 1,
+                        renderTask(tasks.get(i)))
+                    )
+            ).toArray(String[]::new)
```
I think this method body is a bit confusing, would it be better to split it up into several intermediate steps instead? :D

wei2912 replied:
This makes sense, I'll create an intermediate variable for constructing the array of task strings. Thanks for the suggestion!

feliciaivane reviewed (commented):
LGTM, I feel that overall the code is quite easy to read :D Good work!

wei2912 committed:
Implement rudimentary JavaFX GUI

wei2912 committed:
Merge branch 'branch-Level-10'

wei2912 committed:
Duke: Add Assertions

wei2912 committed:
Shift tasks into separate package

wei2912 committed:
Shift GUI-related classes to separate package

wei2912 committed:
Clean up code

A couple of code quality enhancements were carried out:
1. Use of final whenever possible
2. Explicit type declaration where appropriate, with the exception of
   simple variables
3. Simplifying code logic into multiple steps
4. Reduce code duplication

wei2912 committed:
Fix checkstyle violations

wei2912 committed:
Merge pull request #1 from wei2912/branch-A-Assertions

Duke: Add Assertions

wei2912 committed:
Merge branch 'master' into branch-A-CodeQuality

wei2912 committed:
Merge pull request #2 from wei2912/branch-A-CodeQuality

Duke: Code quality enhancements

wei2912 committed:
Parser: Enforce one event per day

This commit enforces the restriction that no two events can be on the
same day.

Some small changes were made to Storage and TaskList so that
the current task list can be used to check for the validity of future
tasks.

wei2912 committed:
Parser: Remove redundant newlines

wei2912 committed:
Merge branch 'branch-BCD-Extension'

wei2912 committed:
DialogBox: Prevent truncation of height

wei2912 committed:
Merge branch 'branch-A-BetterGui'

wei2912 committed:
docs: Add Ui.png

wei2912 committed:
docs: Add User Guide

wei2912 committed:
Merge branch 'branch-A-UserGuide'

wei2912 committed:
build.gradle: Fix typo in Main-Class attribute
